### PR TITLE
ROX-25060: Update info for admission controller timeout

### PR DESCRIPTION
**Pull request by kcarmichael08**
Version(s):
4.5+

[Issue](https://issues.redhat.com/browse/ROX-25060)

Links to docs previews:

[79149--ocpdocs-pr.netlify.app](https://79149--ocpdocs-pr.netlify.app/)
[79149--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/installing_cloud_ocp/install-secured-cluster-cloud-ocp.html](https://79149--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/installing_cloud_ocp/install-secured-cluster-cloud-ocp.html)
[79149--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/installing_cloud_other/install-secured-cluster-cloud-other.html](https://79149--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/installing_cloud_other/install-secured-cluster-cloud-other.html)
[79149--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_ocp/install-secured-cluster-config-options-ocp.html](https://79149--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_ocp/install-secured-cluster-config-options-ocp.html)
[79149--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_ocp/install-secured-cluster-ocp.html](https://79149--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_ocp/install-secured-cluster-ocp.html)
[79149--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_other/install-secured-cluster-other.html](https://79149--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_other/install-secured-cluster-other.html)

QE review: (**ACS has no QE, reviewed/approved by SME**)
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/secured-cluster-configuration-options-operator.adoc [hunk 1]
```diff
@@ -79,8 +79,7 @@ a| Use one of the following values to specify if the admission controller must c
 The default value is `DoNotScanInline`.
 
 | `admissionControl.timeoutSeconds`
-| Use this parameter to specify the maximum number of seconds {product-title} must wait for an admission review before marking it as fail open.
-
+| Use this parameter to specify the maximum number of seconds {product-title-short} must wait for an admission review before marking it as fail open. Beginning in release 4.5, Red{nbsp}Hat reduced the default timeout setting for the {product-title-short} admission controller webhooks from 20 seconds to 10 seconds, resulting in an effective timeout of 12 seconds within the `ValidatingWebhookConfiguration`. This change does not affect {ocp} users because {ocp} caps the timeout at 13 seconds. However, for Kubernetes clusters, image scanning and pulling that are done as part of the webhook execution might require you to configure a longer timeout value.
```
Comment (mclasmeier) [inline modules/secured-cluster-configuration-options-operator.adoc:82]: I am not sure if the word `must` here (as in "{product-title-short} must wait") is a good fit.
I mean, this this a timeout setting. So, it is a guarantee for the waiting time to not exceed the configured timeout. But it is not the case that ACS "must" wait for such a duration(?). (I know that this is also in the old version)

**Comment by mclasmeier** [comment]
Regarding updating the installation instructions for manifest based installs:

I am looking specifically at https://79149--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_other/install-secured-cluster-other#manifest-roxctl_install-secured-cluster-other and noticed that we currently don't go into the details at all? Seems to be only a very high-level explanation of the procedure without diving into the topic of configuring the sensor installation bundle using roxctl command line parameters.

Given this status quo I would say we don't need to update anything there? I would say the same regarding the two places where you added comments. (Especially given that information on this change can be found in the release notes.)

File: modules/secured-cluster-services-config.adoc [hunk 2]
```diff
@@ -155,9 +155,7 @@ This option corresponds to the *Contact image scanners* option in the {product-t
 //TODO: Link to admission controller docs
 
 | `admissionControl.dynamic.timeout`
-| The maximum time, in seconds, {product-title} should wait while evaluating admission review requests.
-Use this to set request timeouts when you enable image scanning.
-If the image scan runs longer than the specified time, {product-title} accepts the request.
+| Use this parameter to specify the maximum number of seconds {product-title-short} must wait for an admission review before marking it as fail open. Beginning in release 4.5, Red{nbsp}Hat reduced the default timeout setting for the {product-title-short} admission controller webhooks from 20 seconds to 10 seconds, resulting in an effective timeout of 12 seconds within the `ValidatingWebhookConfiguration`. This change does not affect {ocp} users because {ocp} caps the timeout at 13 seconds. However, for Kubernetes clusters, image scanning and pulling that are done as part of the webhook execution might require you to configure a longer timeout value.
```
[the remark this answers is not in the document]
Comment (kcarmichael08) [reply modules/secured-cluster-services-config.adoc]: Added conditional text to address this.

File: modules/secured-cluster-configuration-options-operator.adoc [hunk 3]
```diff
@@ -79,8 +79,7 @@ a| Use one of the following values to specify if the admission controller must c
 The default value is `DoNotScanInline`.
 
 | `admissionControl.timeoutSeconds`
-| Use this parameter to specify the maximum number of seconds {product-title} must wait for an admission review before marking it as fail open.
-
+| Use this parameter to specify the maximum number of seconds {product-title-short} must wait for an admission review before marking it as fail open. Beginning in release 4.5, Red{nbsp}Hat reduced the default timeout setting for the {product-title-short} admission controller webhooks from 20 seconds to 10 seconds, resulting in an effective timeout of 12 seconds within the `ValidatingWebhookConfiguration`. This change does not negatively affect {ocp} users because {ocp} caps the timeout at 13 seconds.
```
Comment (jeana-redhat) [inline modules/secured-cluster-configuration-options-operator.adoc:82]: Existing phrasing, but does "fail open" make sense to the expected reader (or is it explained earlier)?

**Review by jeana-redhat — COMMENTED** [review]
Just a terminology question (which applies to both modules). If that's a normal term for this audience please disregard and this is good to go.

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

File: modules/secured-cluster-configuration-options-operator.adoc (hunk 3 above)
Comment (kcarmichael08) [reply modules/secured-cluster-configuration-options-operator.adoc:82]: That's a good question - it is an existing term and I don't see it anywhere else. I'll see if I can find out!

Comment (mclasmeier) [reply modules/secured-cluster-configuration-options-operator.adoc]: Just for the record, the official Kubernetes documentation on this topic can be found here: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy

They use `fail-open` (with a dash) in these docs.

Comment (kcarmichael08) [reply modules/secured-cluster-configuration-options-operator.adoc]: OK, based on some slack research and the link above for the K8s docs, I've added a short description that I hope is accurate. I looked and old OpenShift docs use "fail open" (no hyphen) so I guess I will match those.

Comment (jeana-redhat) [reply modules/secured-cluster-configuration-options-operator.adoc]: reads well to me - looks like the PR needs a refresh

**Pull request merged**